torch/_inductor/cpp_builder.py

```diff
@@ -581,59 +581,25 @@ def _get_os_related_cpp_cflags(cpp_compiler: str) -> list[str]:
     return cflags
 
 
-def _get_ffast_math_flags() -> list[str]:
-    # ffast-math is equivalent to these flags as in
-    # https://github.com/gcc-mirror/gcc/blob/4700ad1c78ccd7767f846802fca148b2ea9a1852/gcc/opts.cc#L3458-L3468
-    # however gcc<13 sets the FTZ/DAZ flags for runtime on x86 even if we have
-    # -ffast-math -fno-unsafe-math-optimizations because the flags for runtime
-    # are added by linking in crtfastmath.o. This is done by the spec file which
-    # only does globbing for -ffast-math.
-    flags = [
-        "fno-trapping-math",
-        "funsafe-math-optimizations",
-        "ffinite-math-only",
-        "fno-signed-zeros",
-        "fno-math-errno",
-    ]
-
-    if is_gcc():
-        flags.append("fexcess-precision=fast")
-
-    return flags
-
-
-def _get_optimization_cflags(
-    cpp_compiler: str, min_optimize: bool = False
-) -> list[str]:
+def _get_optimization_cflags(min_optimize: bool = False) -> list[str]:
+    wrapper_opt_level = (
+        config.aot_inductor.compile_wrapper_opt_level if min_optimize else "O3"
+    )
     if _IS_WINDOWS:
-        return ["O1" if min_optimize else "O2"]
+        return [wrapper_opt_level]
     else:
-        wrapper_opt_level = config.aot_inductor.compile_wrapper_opt_level
         cflags = (
             ["O0", "g"]
             if config.aot_inductor.debug_compile
-            else [wrapper_opt_level if min_optimize else "O3", "DNDEBUG"]
+            else [wrapper_opt_level, "DNDEBUG"]
         )
-        cflags += _get_ffast_math_flags()
-        cflags.append("fno-finite-math-only")
-        if not config.cpp.enable_unsafe_math_opt_flag:
-            cflags.append("fno-unsafe-math-optimizations")
         cflags.append(f"ffp-contract={config.cpp.enable_floating_point_contract_flag}")
 
-        if sys.platform != "darwin":
-            # on macos, unknown argument: '-fno-tree-loop-vectorize'
-            if _is_gcc(cpp_compiler):
-                cflags.append("fno-tree-loop-vectorize")
-            # https://stackoverflow.com/questions/65966969/why-does-march-native-not-work-on-apple-m1
-            # `-march=native` is unrecognized option on M1
-            if not config.is_fbcode():
-                if platform.machine() == "ppc64le":
-                    cflags.append("mcpu=native")
-                else:
-                    cflags.append("march=native")
-
-        if config.aot_inductor.enable_lto and _is_clang(cpp_compiler):
-            cflags.append("flto=thin")
+        if not config.is_fbcode():
+            if platform.machine() == "ppc64le":
+                cflags.append("mcpu=native")
+            else:
+                cflags.append("march=native")
 
         return cflags
```

Comment:
This section got pretty cleaned up. I removed the `-ffast-math` related flags, which didn't make any difference in benchmarks and made code potentially more incorrect. I was also able to remove handling for flags on MacOS that we no longer need on current MacOS compiler versions.

LTO handling got moved to a different location.